daemon.go

```diff
@@ -311,38 +379,63 @@ func (d *daemon) runPeerCheck(ctx context.Context, ma multiaddr.Multiaddr, ai *p
 		PeerFoundInDHT:               addrMap,
 	}
 
+	// FIXME: without AcceleratedDHT client, we usually timeout the full
+	// operation context in the DHT steps. Provide early exit in that
+	// case.
+	if err := ctx.Err(); err != nil {
+		return out, err
+	}
+
+	httpInfo, otherInfo := network.SplitHTTPAddrs(ai)
```

Comment:
A bit more descriptive
```suggestion
	httpInfo, libp2pInfo := network.SplitHTTPAddrs(ai)
```